### PR TITLE
langgraph[patch]: fix annotations in inherited tool schemas

### LINKED ISSUE #2220: ToolNode doesn't detect `InjectedState` annotations for tools with Pydantic `args_schema`
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the [LangGraph](https://langchain-ai.github.io/langgraph/)/LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangGraph/LangChain rather than my code.
- [X] I am sure this is better as an issue [rather than a GitHub discussion](https://github.com/langchain-ai/langgraph/discussions/new/choose), since this is a LangGraph bug and not a design question.

### Example Code

```python
.
```

### Error Message and Stack Trace (if applicable)

```shell
.
```

### Description

.

### System Info

.

### PR DESCRIPTION
**Pull request by baskaryan**
fix #2220

### CODE REVIEW

File: libs/langgraph/langgraph/prebuilt/tool_node.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
 from langchain_core.runnables.utils import Input
 from langchain_core.tools import BaseTool, InjectedToolArg
 from langchain_core.tools import tool as create_tool
+from langchain_core.tools.base import _get_all_basemodel_annotations
```
Comment (vbarda) [inline libs/langgraph/langgraph/prebuilt/tool_node.py:37]: would be nice to make this a public API

**Review by vbarda — APPROVED** [review]

File: libs/langgraph/langgraph/prebuilt/tool_node.py (hunk 1 above)
Comment (baskaryan) [inline libs/langgraph/langgraph/prebuilt/tool_node.py:37]: ah need to check when this was added to core

Comment (baskaryan) [reply libs/langgraph/langgraph/prebuilt/tool_node.py:37]: might need to just copy it over if we don't want to bump core dep

Comment (vbarda) [reply libs/langgraph/langgraph/prebuilt/tool_node.py:37]: i think it's fine to bump core dep

File: libs/langgraph/pyproject.toml [hunk 2]
```diff
@@ -9,7 +9,7 @@ repository = "https://www.github.com/langchain-ai/langgraph"
 
 [tool.poetry.dependencies]
 python = ">=3.9.0,<4.0"
-langchain-core = ">=0.2.39,<0.4"
+langchain-core = "^0.2.42 || ^0.3.14"
```
Comment (vbarda) [inline libs/langgraph/pyproject.toml:12]: TIL

**Review by vbarda — APPROVED** [review]

File: libs/langgraph/pyproject.toml (hunk 2 above)
Comment (baskaryan) [reply libs/langgraph/pyproject.toml:12]: me too

**Pull request merged**